assets/js/base/context/providers/cart-checkout/checkout-processor.ts

```diff
@@ -157,8 +157,25 @@ const CheckoutProcessor = () => {
 		currentRedirectUrl.current = redirectUrl;
 	}, [ billingAddress, shippingAddress, redirectUrl ] );
 
+	// Get shipping rates validation errors from the store.
+	const { shippingRatesValidationError } = useSelect( ( select ) => {
+		const store = select( VALIDATION_STORE_KEY );
+		return {
+			shippingRatesValidationError: store.getValidationError(
+				'shipping-rates-error'
+			),
+		};
+	}, [] );
+
 	const checkValidation = useCallback( () => {
 		if ( hasValidationErrors() ) {
+			// If there is a shipping rates validation error, return the error message to be displayed.
+			if ( shippingRatesValidationError !== undefined ) {
```

Comment:
Since this is within a callback and `shippingRatesValidationError` is not used anywhere else, could we move the lookup of this error inline?

e.g. `select( VALIDATION_STORE_KEY ).getValidationError( 'shipping-rates-error' ) !== undefined`

Might be cleaner?

Reply:
That's a great suggestion. I have updated the code.